### ISSUE TITLE
Modrinth launcher lags and loads one of the cores at 100% on Linux  

### ISSUE THREAD

**Issue by Rikopara** [modrinth/code#1216]
### Describe the bug

I've tried both flatpak and .deb versions. Whenever I scroll or click, it's really laggy and sometimes even freezes until I end the process manually. One of CPU cores (always different) starts to load at 100% until I stop scrolling.

https://github.com/modrinth/theseus/assets/48817751/2dd36797-be0a-40d3-861a-fa3970bf0610

### System information

OS: Zorin OS 17.1 Pro 64 bit
CPU: Intel® Xeon(R) CPU E5-2660 v3 @ 2.60GHz × 20
GPU: RTX 3060
RAM: 32GB

Kernel: 6.5.0-35-generic
DE: GNOME 43.9

**Comment by ErikMCM** [comment]
I am also experiencing this bug. I notice that practically any action causes this behavior making me wonder if it's an animation issue, since they seem to be stuttering. Opening the logs tab crashes the app as well. Since this is open source, I may do some tinkering around with it and see if I can find the root cause...

OS: Kubuntu 22.04.04
CPU: AMD Ryzen 9 5900X (12c24t @ 3.7GHz)
GPU: RTX 3070
RAM: 32GB DDR4 @ 3200 MHz

Kernel: 6.5.0-35-generic
DE: Plasma 5.24.7 (X11)

https://github.com/modrinth/theseus/assets/70036542/e203b61f-7d8c-4f43-8cf2-0e3ab8980950

**Comment by eneshamza** [comment]
I have the same problem. Actually, scrolling does not bother me much, but the button animations are very slow.

OS: Ubuntu 24.04 LTS
CPU: AMD Ryzen 5 5500
GPU: Intel Arc A750
RAM: 16GB DDR4

Kernel: 6.8.0-35-generic
DE: Gnome 46 (X11)

[Ekran kaydı - 2024-06-15 10-21-33.webm](https://github.com/modrinth/theseus/assets/89846189/52b1402d-9e2f-49c8-9816-53bf7b0335c4)

**Comment by eneshamza** [comment]
I have the same problem. Actually, scrolling does not bother me much, but the button animations are very slow.

OS: Ubuntu 24.04 LTS
CPU: AMD Ryzen 5 5500
GPU: Intel Arc A750
RAM: 16GB DDR4

Kernel: 6.8.0-35-generic
DE: Gnome 46 (X11)


https://github.com/modrinth/theseus/assets/89846189/a948a54e-ef95-482f-b679-4730eb463846

**Comment by monomere** [comment]
I have a similar issue! I just updated hoping this issue would be solved haha

OS: Arch Linux
CPU: Intel Core i3-12100
GPU: (iGPU) UHD Graphics 730
RAM: 16 GB
DE: Gnome (Wayland)

**Comment by monomere** [comment]
Indeed, as mentioned in https://github.com/tauri-apps/tauri/issues/6200, building from source (in both debug and release profiles) fixed the scrolling issue.

**Comment by CommanderGL** [comment]
Weird as when I use the `modrinth-app` aur package which I assume is building from source I still get terrible scrolling.

**Comment by Sunoctagram** [comment]
Experiencing the same issue.

OS: EndeavorOS Linux
CPU: AMD Ryzen 7 2700 Eight-Core Processor
GPU: NVIDIA GTX 1050
RAM: 32 GB
DE: KDE plasma

**Comment by Kerkpower** [comment]
i have the same issue, particularly when scrolling on pages with a lot of items

OS: EndeavourOS Linux
CPU: AMD Ryzen 5 5600G
GPU: (APU) AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
RAM: 16 GB
DE: Plasma 6.1.1 (wayland)

**Comment by louiszn** [comment]
Same issue here

OS: Arch Linux
CPU: Intel i3-7100U
GPU: Intel HD Graphics 620
RAM: 8 GB
DE: Plasma 6.1.2 (Wayland)

**Comment by NotMurPh** [comment]
same issues here
Package: modrinth-app-bin from aur

**Comment by SMASTER4** [comment]
Intrestingly enough does the problem only accure on the first few pages :thinking:
IDK, but I feel like the popularity of projects could affect the performance
Like for real it could be releated to some issues with bigger data types in tauri(unlikely)

**Comment by AsaTheUncle** [comment]
Same issue here. Although I also get the crash from opening the Logs tab (https://github.com/modrinth/code/issues/1127) 
Package: modrinth-app from aur
OS: Manjaro Linux
DE: KDE Plasma 6.1.4 w/ Wayland

**Comment by negativems** [comment]
Same here

OS: Endeavour (6.11.2-arch1-1)
CPU: AMD Ryzen 7 7800X3D
GPU: RX 7800 XT
RAM: 32 GB DDR5
DE: bspwm

**Comment by Domikulas** [comment]
Same here

OS: Arch
CPU: AMD Ryzen 5 2400G
GPU: AMD Radeon RX 590
RAM: 16 GB DDR4
DE: KDE Plasma 6.1.5

I don't want to build it from source. Are there any solutions yet?

**Comment by x-0D** [comment]
Same issue on SteamDeck OLED 1TB (SteamOS 3.6)

Very Laggy UI, one CPU core 100% usage.

**Comment by sinzek** [comment]
Issue still persists on linux mint (Cinnamon)

**Comment by pokahs** [comment]
Also having this issue on debian 12... any solutions?

**Comment by Rikopara** [comment]
@pokahs clear modrinth reinstall doesn't help?

**Comment by Rikopara** [comment]
I've noticed that different modrinth packages also behave different. For me (Fedora 41) RPM works just fine, but appimage is extremely laggy. So, as possible solution, you may try every package your system supports, or even build modrinth from source. Hope this helps.

**Comment by TheMightyMcGrew** [comment]
also having this issue with Pop!_OS. Randomly chooses 1 core and shoots to the moon at 100%![Image](https://github.com/user-attachments/assets/f9399583-2af2-48e7-8265-a0a81f6279dd)
AMD Ryzen 9 5900X 12-Core Processor
64GB DDR4
NVIDIA GeForce RTX 3070 Ti
Pop!_OS 22.04 LTS

**Comment by ArnaudCHEVALME** [comment]
Still have the same issue.
OS: Arch
packages tested : aur/modrinth-app, aur/modrinth-app-git, aur/modrinth-app-bin

![Image](https://github.com/user-attachments/assets/615962e8-d1fb-4246-a53a-48b884618a03)

**Comment by switchmaxfx** [comment]
Just want to say im having the same issue. using modrinth-app-bin from aur

**Comment by aperiogon3141** [comment]
Facing the same difficulty

**Comment by PoProstuWitold** [comment]
So is it really unresolved so far?

**Comment by aperiogon3141** [comment]
adding WEBKIT_DISABLE_DMABUF_RENDERER=0 to the launch command in /usr/bin/modrinth-app worked for me.

**Comment by TheMightyMcGrew** [comment]
I found it easier just to switch to .deb

**Comment by Android789515** [comment]
Same issue.  And devs have not commented in over a year.  Makes me wonder if they even care about their app.

**Comment by Rikopara** [comment]
@Android789515 
I switched to the Prism launcher.

**Comment by Jab11n** [comment]
Same issue here on Kubuntu 24.04 with Modrinth App installed via .deb.

**Comment by aperiogon3141** [comment]
> Same issue here on Kubuntu 24.04 with Modrinth App installed via .deb.

adding WEBKIT_DISABLE_DMABUF_RENDERER=0 to the launch command in /usr/bin/modrinth-app worked for me. try it out

**Comment by PhantomFilm** [comment]
> > Same issue here on Kubuntu 24.04 with Modrinth App installed via .deb.
> 
> adding WEBKIT_DISABLE_DMABUF_RENDERER=0 to the launch command in /usr/bin/modrinth-app worked for me. try it out

Im completely new to Linux im using Mint, how do I get into the file to edit it? I click and it opens it, I followed the same path.

**Comment by Rikopara** [comment]
@PhantomFilm Test it out though terminal, navigate to `/usr/bin` and type `./ModrinthApp WEBKIT_DISABLE_DMABUF_REBDERER=0`

If this works, edit your .desktop entry that you use to open Modrinth (or just add new one). I suggest you to install "Pins" app from flathub since you're new to Linux.

**Comment by PhantomFilm** [comment]
_

https://github.com/user-attachments/assets/a96d7a04-e82d-45e7-a8d1-25aa908236e6

_

@Rikopara What am I doing wrong?

**Comment by PhantomFilm** [comment]
https://github.com/user-attachments/assets/92e96cca-3ef3-4330-a92a-6e2f99887fc5

@Rikopara I tried this too

**Comment by Rikopara** [comment]
Oh shoot I lied, this is env variable so you need to type `WEBKIT_DISABLE_DMABUF_RENDERER=0  /usr/bin/ModrinthApp`
It still may not work tho. 
P.S. : google about `cd` syntax, looks like you don't know how to navigate directories in terminal.

This issue is stale so I'm closing it.

**Comment by Alshat** [comment]
Same issue here. I wasn't able to performed the suggested changes.

OS: Zorin OS 17.3 Core
CPU: Intel® Core™ i5-6600K CPU
GPU: RTX 2070
RAM: 16 GB DDR4
DE/WM: X11

**Comment by negativems** [comment]
I fixed it by running directly the executable from `/opt/modrinth-app/modrinth-app`.

So I changed the .desktop (`/usr/share/applications/modrinth-app.desktop`) to run this path instead of running `modrinth-app` and I dont know why `/usr/bin/modrinth-app` does have this issue.

P.S. This paths should be the same if you are on Arch and installed "modrinth-app-bin" package.
